test/nullable.jl

```diff
@@ -388,6 +393,119 @@ end
 @test Base.promote_op(-, Nullable{Float64}, Nullable{Int}) == Nullable{Float64}
 @test Base.promote_op(-, Nullable{DateTime}, Nullable{DateTime}) == Nullable{Base.Dates.Millisecond}
 
+# filter
+for p in (_ -> true, _ -> false)
+    @test @inferred(filter(p, Nullable()))      |> isnull_oftype(Union{})
+    @test @inferred(filter(p, Nullable{Int}())) |> isnull_oftype(Int)
+end
+@test @inferred(filter(_ -> true, Nullable(85)))  === Nullable(85)
+@test @inferred(filter(_ -> false, Nullable(85))) |> isnull_oftype(Int)
```

Comment:
I think I mentioned this before, but for `isbits` types you should check that the fast-path was taken by doing `=== Nullable(85, false)`. This also applies to null-safe operations like `-`, `.+` and `.-` below, and to the `(eop, op)` loop when `null_safe_op` is true.